src/typesConverter.ts

```diff
@@ -334,7 +338,15 @@ export const parseSchemas = ({ json, swaggerVersion, overrideSchemas }: GetSchem
                  * Is schema is a simple object or is it extends from another schema
                  */
                 if (schema[SwaggerProps.Type] === DataTypes.Object || schema[SwaggerProps.AllOf]) {
-                    result += parseObject({ schema, schemaKey });
+                    /**
+                     * Sometimes in swagger v2 schema key could be named as SomeDto[AnotherDto]
+                     */
+                    if (swaggerVersion === 2 && schemaKey.includes('[') && schemaKey.includes(']')) {
+                        const strings = schemaKey.split('[');
+                        result += parseObject({ schema, schemaKey: strings[0] });
+                    } else {
+                        result += parseObject({ schema, schemaKey });
+                    }
```

Comment:
I think you can always call `schemaKey.split('[')` as even if the `schemaKey` doesn't contain any `[`, the first item in the array will work:
```javascript
'SomeDto'.split('[')
// ["SomeDto"]

'SomeDto[AnotherDto]'.split('[')
// ["SomeDto", "AnotherDto]"]
```

Reply:
@ardeois You are right , but currently this issue was found only in Swagger v2. We didn't faced such issues in v3. I want to be clear why we added `if` statement here and when it can happened . If such issue will occur in v3 - sure I could easily refactor this part of code.

Reply:
ok makes sense to keep the if then